### PR TITLE
Feature/update plugin for action menu

### PR DESCRIPTION
**Pull request by k-y0u**
Update PluginEvent to add component action menu management: Select, Edit, Delete.
Update ModelView to manage component deletion.

### CODE REVIEW

File: src/components/ModelizerModelView.vue [hunk 1]
```diff
@@ -25,21 +27,61 @@ import ComponentDefinitionsDrawer from 'src/components/drawer/ComponentDefinitio
...
-  pluginSubscription = PluginEvent.subscribe((event) => {
-    if (event === 'ready') {
-      data.plugins = getPlugins();
+/**
+ * Delete a component from a tree of components
+ *
+ * @param {Object} component - component to find and remove.
+ * @return {Boolean} true if component was found and delete otherwise false.
+ */
+function deleteComponent(component, tree) {
```
Comment (Zorin95670) [inline src/components/ModelizerModelView.vue:42]: put this in `composable/PluginManager.js`

File: src/components/ModelizerModelView.vue [hunk 2]
```diff
@@ -25,21 +27,61 @@ import ComponentDefinitionsDrawer from 'src/components/drawer/ComponentDefinitio
...
     }
-  });
+    return false;
+  }
+
+  tree.splice(index, 1);
+  return true;
+}
+
+function deletePluginComponentAndRedraw(component) {
```
Comment (Zorin95670) [inline src/components/ModelizerModelView.vue:62]: rename `component` to `event`

File: src/components/ModelizerModelView.vue [hunk 3]
```diff
@@ -25,21 +27,61 @@ import ComponentDefinitionsDrawer from 'src/components/drawer/ComponentDefinitio
...
-    if (event === 'ready') {
-      data.plugins = getPlugins();
+/**
+ * Delete a component from a tree of components
+ *
+ * @param {Object} component - component to find and remove.
+ * @return {Boolean} true if component was found and delete otherwise false.
+ */
+function deleteComponent(component, tree) {
+  const index = tree.findIndex((c) => c.id === component.id);
```
Comment (Zorin95670) [inline src/components/ModelizerModelView.vue:43]: replace: `tree.findIndex(({ id }) => id === component.id)`

File: src/components/ModelizerModelView.vue [hunk 4]
```diff
@@ -25,21 +27,61 @@ import ComponentDefinitionsDrawer from 'src/components/drawer/ComponentDefinitio
...
+
+  if (index === -1) {
+    const treeWithChildren = tree.filter((c) => c.children && c.children.length > 0);
```
Comment (Zorin95670) [inline src/components/ModelizerModelView.vue:46]: same type of replacement here

File: src/components/ModelizerModelView.vue [hunk 5]
```diff
@@ -25,21 +27,61 @@ import ComponentDefinitionsDrawer from 'src/components/drawer/ComponentDefinitio
...
+
+    for (let i = 0; i < treeWithChildren.length; i += 1) {
```
Comment (Zorin95670) [inline src/components/ModelizerModelView.vue:48]: rename i to index

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: src/components/ModelizerModelView.vue [hunk 6]
```diff
@@ -25,21 +27,61 @@ import ComponentDefinitionsDrawer from 'src/components/drawer/ComponentDefinitio
...
+      const haveBeenDeleted = deleteComponent(component, treeWithChildren[i].children);
```
Comment (justinehell) [inline src/components/ModelizerModelView.vue:49]: `haveBeenDeleted` -> `hasBeenDeleted`

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/components/ModelizerModelView.vue (hunk 5 above)
Comment (k-y0u) [reply src/components/ModelizerModelView.vue:48]: not possible

File: src/components/ModelizerModelView.vue (hunk 1 above)
Comment (k-y0u) [reply src/components/ModelizerModelView.vue]: done

File: src/components/ModelizerModelView.vue (hunk 2 above)
Comment (k-y0u) [reply src/components/ModelizerModelView.vue]: done

File: src/components/ModelizerModelView.vue (hunk 3 above)
Comment (k-y0u) [reply src/components/ModelizerModelView.vue]: done

File: src/components/ModelizerModelView.vue (hunk 4 above)
Comment (k-y0u) [reply src/components/ModelizerModelView.vue]: done

File: src/components/ModelizerModelView.vue (hunk 6 above)
Comment (k-y0u) [reply src/components/ModelizerModelView.vue]: done

File: src/components/ModelizerModelView.vue (hunk 5 above)
Comment (k-y0u) [reply src/components/ModelizerModelView.vue]: done

File: src/components/ModelizerModelView.vue [hunk 7]
```diff
@@ -34,42 +37,20 @@ const data = reactive({
 });
 
 /**
- * Delete a component from a tree of components
- *
- * @param {Object} component - component to find and remove.
- * @return {Boolean} true if component was found and delete otherwise false.
+ * Search a component to delete amongs all the components of all the plugins.
```
Comment (justinehell) [inline src/components/ModelizerModelView.vue:40]: `amongs` -> `among` or `amongst` (meaning is the same, amongst is more formal)

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/components/ModelizerModelView.vue (hunk 7 above)
Comment (k-y0u) [reply src/components/ModelizerModelView.vue]: Done

**Review by justinehell — APPROVED** [review]

File: src/composables/PluginManager.js [hunk 8]
```diff
@@ -154,3 +154,22 @@ export function getPlugins() {
 export function getPluginByName(name) {
   return instanciatePlugins.find((p) => p.name === name);
 }
+
+/**
+ * Delete a component from a tree of components
```
Comment (Zorin95670) [inline src/composables/PluginManager.js:159]: missing `.`

File: src/composables/PluginManager.js [hunk 9]
```diff
@@ -154,3 +154,22 @@ export function getPlugins() {
...
+ *
+ * @param {Object} component - component to find and remove.
```
Comment (Zorin95670) [inline src/composables/PluginManager.js:161]: You don't receive a component here. Please rename to Event and update the documenation like:
```
* @param {Object} event - Event that indicate all details of componnet to remve
* @param {String} event.id - Component id to remove
```

File: src/components/ModelizerModelView.vue [hunk 10]
```diff
@@ -22,14 +24,30 @@ import {
...
 
 let pluginInitSubscription;
+let pluginDeleteSubscription;
 const data = reactive({
   plugins: [],
 });
 
+/**
+ * Search a component to delete among all the components of all the plugins.
+ * @param {Object} event Object containing id of the component to delete.
```
Comment (Zorin95670) [inline src/components/ModelizerModelView.vue:41]: `event - Object`

**Review by Zorin95670 — CHANGES_REQUESTED** [review]

File: src/components/ModelizerModelView.vue (hunk 10 above)
Comment (k-y0u) [reply src/components/ModelizerModelView.vue]: Done

File: src/composables/PluginManager.js (hunk 9 above)
Comment (k-y0u) [reply src/composables/PluginManager.js]: Done (better)

File: src/composables/PluginManager.js (hunk 8 above)
Comment (k-y0u) [reply src/composables/PluginManager.js]: done

**Review by justinehell — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Pull request merged**